encoding.go

```diff
@@ -69,12 +69,22 @@ func ParseNode(serialized []byte, depth byte, comm []byte) (VerkleNode, error) {
 		ln.Commit()
 		return ln, nil
 	case internalRLPType:
-		return deserializeIntoStateless(serialized[1:33], serialized[33:], depth, comm)
+		return CreateInternalNode(serialized[1:33], serialized[33:], depth, comm)
 	default:
 		return nil, ErrInvalidNodeEncoding
 	}
 }
 
+func ParseStatelessNode(serialized []byte, depth byte, comm []byte) (VerkleNode, error) {
+	if len(serialized) < 64 {
+		return nil, serializedPayloadTooShort
+	}
+	if serialized[0] == internalRLPType {
+		return deserializeIntoStateless(serialized[1:33], serialized[33:], depth, comm)
+	}
+	return nil, ErrInvalidNodeEncoding
+}
+
```

Comment:
That's nice, but is only able to desserialize a statless, internal node. That's a problem In the case of the stateless insert, for example, because hashnode resolution in stalessnode insert should return stateless internal nodes. So the best way would be to make a function that has the old behavior, including being able to return `LeafNode` when deserializing a leaf.

I understand it's code duplication, but it's possible to use a helper function to reduce the amount of duplicated code, and hopefully `StatelessNode` is going to disappear eventually.